docs/src/literate-tutorials/dg_heat_equation.jl

```diff
@@ -0,0 +1,411 @@
+# # [Discontinuous Galerkin Heat equation](@id tutorial-dg-heat-equation)
+#
+# ![](dg_heat_equation.png)
+#
+# *Figure 1*: Temperature field on the unit square with an internal uniform heat source
+# solved with inhomogeneous Dirichlet boundary conditions on the left and right boundaries and flux on the top and bottom boundaries.
+#
+#-
+#md # !!! tip
+#md #     This example is also available as a Jupyter notebook:
+#md #     [`dg_heat_equation.ipynb`](@__NBVIEWER_ROOT_URL__/examples/dg_heat_equation.ipynb).
+#-
+#
+# This example was developed
+# as part of the Google summer of code funded project
+# ["Discontinuous Galerkin Infrastructure For the finite element toolbox Ferrite.jl"](https://summerofcode.withgoogle.com/programs/2023/projects/SLGbRNI5)
+#
+# ## Introduction
+#
+# The heat equation is the "Hello, world!" equation of finite elements.
+# Here we solve the equation on a unit square with a uniform internal source, using the interior penalty formulation.
+# The strong form of the (linear) heat equation is given by
+#
+# ```math
+#  -\nabla \cdot (k \nabla u) = f  \quad \textbf{x} \in \Omega,
+# ```
+#
+# where $u$ is the unknown temperature field, $k$ the heat conductivity,
+# $f$ the heat source and $\Omega$ the domain. For simplicity we set $f = 1$
+# and $k = 1$. We will consider inhomogeneous Dirichlet boundary conditions such that
+# ```math
+# u(\textbf{x}) = 1 \quad \textbf{x} \in \partial \Omega_u^+, \\
+# u(\textbf{x}) = -1 \quad \textbf{x} \in \partial \Omega_u^-,
+# ```
+# and Neumann boundary conditions such that
+# ```math
+# \nabla u(\textbf{x}) \cdot n = 1 \quad \textbf{x} \in \partial \Omega_n^+, \\
+# \nabla u(\textbf{x}) \cdot n = -1 \quad \textbf{x} \in \partial \Omega_n^-,
+# ```
+# where $\partial \Omega$ denotes the boundaries of $\Omega$ characterized by their normals directions
+# as the following:
+#
+# | Boundary              | Normal direction |
+# |-----------------------|------------------|
+# | $\partial \Omega_u^+$ | (1 , 0)          |
+# | $\partial \Omega_u^-$ | (-1 , 0)         |
+# | $\partial \Omega_n^+$ | (0 , 1)          |
+# | $\partial \Omega_n^-$ | (0 , -1)         |
+# The definitions of jumps and averages used in this examples are
+# ```math
+#  \{u\} = \frac{1}{2}(u^+ + u^-),\quad \llbracket u\rrbracket  = u^+ \cdot n^+ + u^- \cdot n^-\\
+# ```
+# !!! details "Derivation of the weak form for homogeneous Dirichlet boundary condition"
```

Comment:
I would like vector-valued quantities in boldface in this derivation to be consistent. 

Also, in this derivation, Gamma is used for the internal/element boundary, while Gamma^0 is used below. Is this intentional, or should this be made the same?

Reply:
Yeah it's intentional, the derivation assumed homogeneous dbc so the face integral isn't split into boundary and interface terms
In the derivation the integral is over Gamma while in the example it's split into an integral over Gamma_0 and another one over Omega_N

Reply:
xref https://github.com/Ferrite-FEM/Ferrite.jl/pull/787#discussion_r1439117189